kerastuner/applications/efficientnet.py

```diff
@@ -0,0 +1,147 @@
+              specified.
+        input_tensor: Keras tensor to use as image input for the model.
+              One of `input_shape` or `input_tensor` must be
+              specified.
+        classes: number of classes to classify images into.
+        augmentation: optional Model or HyperModel for image augmentation.
+        **kwargs: Additional keyword arguments that apply to all
+            HyperModels. See `kerastuner.HyperModel`.
+    """
+    def __init__(self,
+                 input_shape=None,
+                 input_tensor=None,
+                 classes=None,
+                 augmentation=None,
+                 **kwargs):
+
+        super(HyperEfficientNet, self).__init__(**kwargs)
+        if not classes:
+            raise ValueError('You must specify `classes`')
+
+        if input_shape is None and input_tensor is None:
+            raise ValueError('You must specify either `input_shape` '
+                           'or `input_tensor`.')
+
+        self.input_shape = input_shape
+        self.input_tensor = input_tensor
+        self.classes = classes
+        self.augmentation = augmentation
+
+    def build(self, hp):
+
+        if self.input_tensor is not None:
+            inputs = tf.keras.utils.get_source_inputs(self.input_tensor)
+            x = self.input_tensor
+        else:
+            inputs = layers.Input(shape=self.input_shape)
+            x = inputs
+
+        if self.augmentation:
+            if isinstance(self.augmentation, hypermodel.HyperModel):
+                augmentation = self.augmentation.build(hp)
+            elif isinstance(self.augmentation, keras.models.Model):
+                augmentation = self.augmentation
+            else:
+                raise ValueError('Keyword augment is either a HyperModel or a Keras Model')
+            x = augment(x)
+
+        # Select one of pre-trained EfficientNet as feature extractor
+        version = hp.Choice('version', ['B{}'.format(i) for i in range(8)], default='B0')
+        img_size_dict = {'B0': 224,
+                         'B1': 240,
+                         'B2': 260,
+                         'B3': 300,
+                         'B4': 380,
+                         'B5': 456,
+                         'B6': 528,
+                         'B7': 600}
+        img_size = img_size_dict[version]
+
+        x = preprocessing.Resizing(img_size, img_size, interpolation='bilinear')(x)
+        efficientnet_model = EFFICIENTNET_MODELS[version](include_top=False, input_tensor=x)
+
+        # Rebuild top layers of the model.
+        x = efficientnet_model.output
+
+        pooling = hp.Choice('pooling', ['avg', 'max'], default='avg')
+        if pooling == 'avg':
+            x = layers.GlobalAveragePooling2D(name='avg_pool')(x)
+        elif pooling == 'max':
+            x = layers.GlobalMaxPooling2D(name='max_pool')(x)
+
+        top_dropout_rate = hp.Float('top_dropout_rate', 0.2, 0.8, step=0.2, default=0.2)
```

Comment:
can u put the arg name for 0.2 and 0.8 here to make the code more readable?

Reply:
Thanks for the suggestion. Added arg name.